diff --git a/core/Makefile b/core/Makefile
index b7f92ee..34e1178 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -803,6 +803,10 @@ $(TARGET_RECOVERY_ROOT_TIMESTAMP): $(INTERNAL_RECOVERY_FILES) \
 	mkdir -p $(TARGET_RECOVERY_ROOT_OUT)/tmp
 	echo Copying baseline ramdisk...
 	cp -R $(TARGET_ROOT_OUT) $(TARGET_RECOVERY_OUT)
+ifneq ($(BOARD_RECOVERY_RAMDISK_EXTRA_SCRIPT),)
+	@echo extra recovery ramdisk step...
+	-$(BOARD_RECOVERY_RAMDISK_EXTRA_SCRIPT)
+endif
 ifneq ($(BOARD_USES_COMBINED_RECOVERY),true)
 	rm $(TARGET_RECOVERY_ROOT_OUT)/init*.rc
 endif
@@ -823,6 +827,7 @@ endif
 	$(foreach item,$(recovery_fstab), \
 	  cp -f $(item) $(TARGET_RECOVERY_ROOT_OUT)/etc/recovery.fstab)
 	cp $(RECOVERY_INSTALL_OTA_KEYS) $(TARGET_RECOVERY_ROOT_OUT)/res/keys
+
 ifeq ($(BOARD_USES_COMBINED_RECOVERY),true)
 	cp $(INSTALLED_DEFAULT_PROP_TARGET) $(TARGET_RECOVERY_ROOT_OUT)/default.prop
 	rm -rf $(TARGET_RECOVERY_ROOT_OUT)/misc
